docs/features/async-main.md

```diff
@@ -0,0 +1,12 @@
+# Async Task Main
+## [dotnet/csharplang proposal](https://github.com/dotnet/csharplang/blob/master/proposals/async-main.md)
+
+## Technical Details
+
+* The compiler must recognize `Task` and `Task<int>` as valid entrypoint return types in addition to `void` and `int`.
```

Comment:
"It's left for future".   
Some design work was done specifically to not preventing supporting Task-likes in the future if needed.  
For example - the return type is inferred from the shape of GetAwaiter and not from the `T` in the Task<T>.